src/query-builder/SelectQueryBuilder.ts

```diff
@@ -2027,4 +2031,32 @@ export class SelectQueryBuilder<Entity> extends QueryBuilder<Entity> implements
         return this.queryRunner || this.connection.createQueryRunner("slave");
     }
 
+    protected mightGlobalScoped(): this {
+        if (!this.expressionMap.globalScoped)
+            return this;
+        try {
+            const metadata = this.expressionMap.mainAlias!.metadata;
+            const scopeFns = metadata.scopes.filter(scope => scope.global).map(scope => {
+                return (scope.target as any)[scope.propertyName] as ScopeFn<this>;
+            });
+            this.scope(scopeFns);
+        } catch (_) {
+        }
+        return this;
+    }
+
+    unscoped(globalScoped: boolean = false): this {
+        this.expressionMap.globalScoped = globalScoped;
+        return this;
+    }
+
+    scope(scopeFn: ScopeFn<this>): this;
+    scope(scopeFns: Array<ScopeFn<this>>): this;
+    scope(scopeFn:  ScopeFn<this> | Array<ScopeFn<this>>): this {
+        const scopeFns = Array.isArray(scopeFn) ? scopeFn : [scopeFn];
+        scopeFns.forEach(scopeFn => {
+            scopeFn(this);
```

Comment:
Should provide alias as second parameter - two entities can have the same field name.